compiler/lib/driver.ml

```diff
@@ -104,9 +104,22 @@ let exact_calls profile p =
       | O3 -> false
       | O1 | O2 -> true
     in
-    let info = Global_flow.f ~fast p in
-    Specialize.f ~function_arity:(fun f -> Global_flow.function_arity info f) p
-  else p
+    if Config.Flag.globaldeadcode ()
+    then
+      let p, sentinal = Deadcode_dgraph.add_sentinal p in
+      let info = Global_flow.f ~fast p in
+      let p = Deadcode_dgraph.f p sentinal info in
+      let p =
+        Specialize.f ~function_arity:(fun f -> Global_flow.function_arity info f) p
+      in
+      p, Some sentinal
+    else
+      let info = Global_flow.f ~fast p in
+      let p =
+        Specialize.f ~function_arity:(fun f -> Global_flow.function_arity info f) p
+      in
+      p, None
+  else p, None
```

Comment:
Maybe you can move the call to `Specialize.f` outside of the conditional:
```suggestion
    let (p, info, sentinal) =
      if Config.Flag.globaldeadcode ()
      then
        let p, sentinal = Deadcode_dgraph.add_sentinal p in
        let info = Global_flow.f ~fast p in
        let p = Deadcode_dgraph.f p sentinal info in
        (p, info, Some sentinal)
      else
        let info = Global_flow.f ~fast p in
        (p, info, None)
    in
    let p =
      Specialize.f ~function_arity:(fun f -> Global_flow.function_arity info f) p
    in
    p, Some sentinal
  else p, None
```

Reply:
[a343dc9](https://github.com/ocsigen/js_of_ocaml/pull/1503/commits/a343dc9a6d11d4ab4159f399980168b81aadc082)